lib/azure/core/http/http_request.rb

```diff
@@ -149,7 +154,7 @@ def call
 
           response = HttpResponse.new(res)
           response.uri = uri
-          raise response.error unless response.success?
+          raise response.error if !response.success? && !@has_retry_filter
```

Comment:
The current test code already covers the `retry_policy` in this gem. There is no test need be added because no other policy is implemented in this gem. I've implemented a linear retry policy and exponential policy and their test codes in azure-storage gem. So I would leverage those test cases.